frontend/assignments/2/frontend/script/htmlScript/code.js

```diff
@@ -0,0 +1,587 @@
+    if (tweetContent.length > 0) {
+      let img = null;
+
+      if (isImageUploaded()) {
+        img = document.querySelector(".post-input-image img").src;
+      }
+      console.log(imagePath)
+      const newPost = {
+        username: username,
+        timestamp: new Date().getTime(),
+        message: tweetContent,
+        img: img == null ? null : img,
+      };
+
+      const newPostJSON = JSON.stringify(newPost);
+
+      // Calculate the payload size in bytes
+      const payloadSizeInBytes = new Blob([newPostJSON]).size;
+    
+      console.log("Payload size of newPost object:", payloadSizeInBytes, "bytes");
+      console.log(newPost);
+      fetch(`http://${window.location.hostname}:3000/api/posts`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newPost),
+      })
+        .then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error("Failed to add post");
+          }
+        })
+        .then((post) => {
+          createPostElement(post);
+          if (window.innerWidth < 500) {
+            tweetBoxProfileMobile.style.display = "none";
+            posts.style.display = "block";
+            profileIcon.style.display = "flex";
+            floatingTweetBoxIcon.style.display = "flex";
+            tweetBox.style.display = "none";
+          }
+          window.location.reload();
+
+          tweetInput.value = "";
+          postButton.style.backgroundColor = "";
+          postButton.style.color = "";
+        });
+    }
+  });
+
+ 
+  document.addEventListener("click", function (event) {
+    if (event.target && event.target.classList.contains("retweet")) {
+      const likeIcon = event.target;
+      if (likeIcon.src.includes("retweet.svg")) {
+        likeIcon.src = "images/icons/retweet-blue.svg";
+
+        let likeCountElement =
+          likeIcon.parentElement.querySelector(".tweet-count");
+        if (
+          !likeCountElement ||
+          isNaN(parseInt(likeCountElement.textContent))
+        ) {
+          likeCountElement = document.createElement("div");
+          likeCountElement.classList.add("tweet-count");
+          likeCountElement.textContent = "1";
+          likeIcon.parentElement.appendChild(likeCountElement);
+        } else {
+          likeCountElement.textContent =
+            parseInt(likeCountElement.textContent) + 1;
+        }
+      } else {
+        likeIcon.src = "images/icons/retweet.svg";
+        let likeCountElement =
+          likeIcon.parentElement.querySelector(".tweet-count");
+        if (likeCountElement) {
+          let count = parseInt(likeCountElement.textContent);
+          if (count > 0) {
+            console.log(count);
+            if (count == 1) {
+              likeCountElement.remove();
+            } else {
+              likeCountElement.textContent = count - 1;
```

Comment:
Doesn't it make more sense to update your posts in this scenario (not cutting marks). This is something easy which should have been done.